File: Hacker Ranks/Sales by Match.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the sockMerchant function below.
function sockMerchant(n, ar) {
    const Counter = (array) =>{
    let count = {};
    array.forEach(i => count[i] = (count[i] || 0) + 1);
  return count;
    }
    Counter(ar)
    let pair = (Object.values(Counter(ar)))
    for(let i = 0; i<pair.length; i++) pair[i] = parseInt(pair[i]/2,10)
    return (pair.reduce((x,y)=> x+y))

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    const ar = readLine().split(' ').map(arTemp => parseInt(arTemp, 10));

    let result = sockMerchant(n, ar);

    ws.write(result + "\n");

    ws.end();
}

File: Hacker Ranks/Sales by Match.py
#!/bin/python3

import math
import os
import random
import re
import sys
from collections import Counter
# Complete the sockMerchant function below.
def sockMerchant(n, ar):
    counter = Counter(ar)
    pairsocks = list(counter.values())
    cnt = 0
    for i in range(len(pairsocks)):
        pairsocks[i] = pairsocks[i]//2
    return sum(pairsocks)
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    ar = list(map(int, input().rstrip().split()))

    result = sockMerchant(n, ar)

    fptr.write(str(result) + '\n')

    fptr.close()
